metagraph/src/cli/assemble.cpp

```diff
@@ -149,6 +149,9 @@ void call_masked_graphs(const AnnotatedDBG &anno_graph,
         auto line_split = utils::split_string(line, "\t", false);
 
         if (line[0] == '@') {
+            // lines of the form '@\tIN1,IN2,IN3\tOUT1,OUT2,OUT3' designate
```

Comment:
What if comma is part of a label name?

What about this format:
```
@ Experiment 1: ...
IN: LABEL1
IN: LABEL2
...
OUT: ...
...
@ Experiment 2: ...
# other comments
IN: ...
```

Reply:
I think the best way to avoid all of these issues is to pass in a JSON file to define the experiments. This way, there's no ambiguity in parsing and we can just use the library to read it.

Reply:
But that is less human-readable/writable.

Reply:
It can be just as readable. For example
```json
{
    "shared_labels": [
        {
            "start_index": 0,
            "in": ["IN1","IN2","IN3"],
            "out": ["OUT1","OUT2"]
        }
    ],
    "experiments": [
        {
            "name": "fuzzy",
            "in": ["myIN1","myIN2","myIN3"],
            "out": ["myOUT1","myOUT2"],
            "in_min_fraction": 0.6,
            "out_max_fraction": 0.5,
            "other_max_fraction": 0.0,
            "unitig_in_min_fraction": 1.0,
            "unitig_out_max_fraction": 0.0
        },
        {
            "name": "or",
            "in": ["myIN1","myIN2"],
            "out": ["myOUT1","myOUT2"],
            "in_min_fraction": 0.5,
            "out_max_fraction": 0.0,
            "other_max_fraction": 0.0,
            "unitig_in_min_fraction": 1.0,
            "unitig_out_max_fraction": 0.0
        }
    ]
}
```

Reply:
Yeah, readability looks ok but it looks way harder to write.

Reply:
But maybe it's fine, if you also add an example file to the repo, so there is something to start with.